trl/trainer/dpo_trainer.py

```diff
@@ -1234,9 +1238,27 @@ def get_batch_loss_metrics(
         else:
             ref_chosen_logps, ref_rejected_logps = self.compute_ref_log_probs(batch)
 
-        losses, chosen_rewards, rejected_rewards = self.dpo_loss(
-            model_output["chosen_logps"], model_output["rejected_logps"], ref_chosen_logps, ref_rejected_logps
-        )
+        if "," in self.loss_type:
```

Comment:
Idelament, at this point, we would have
- `self.loss_type` a list of strings (eg, `[“sigmoid”, “bco_pair”]`)
- `self.loss_type_to_weights` a dict of `str, float` which for each loss type associates a weight.

Parsing for loss type could be done directly in the config, as here:

https://github.com/huggingface/trl/blob/fe4b5efe4e23f4331ba9c5b0c8bd92dc8302c287/trl/trainer/nash_md_config.py#L34-L46